tests/test_utils/codec.cpp

```diff
@@ -0,0 +1,103 @@
+#include "test_utils/codec.hpp"
+
+#include <vector>
+
+#include <cbv2g/app_handshake/appHand_Decoder.h>
+#include <cbv2g/app_handshake/appHand_Encoder.h>
+
+#include <cbv2g/din/din_msgDefDecoder.h>
+#include <cbv2g/din/din_msgDefEncoder.h>
+
+#include <cbv2g/iso_20/iso20_AC_Decoder.h>
+#include <cbv2g/iso_20/iso20_AC_Encoder.h>
+#include <cbv2g/iso_20/iso20_DC_Decoder.h>
+#include <cbv2g/iso_20/iso20_DC_Encoder.h>
+
+namespace test_utils {
+
+template <typename DocType>
+static EncodingResult encode(int (*encode_func)(exi_bitstream_t*, DocType*), const DocType& request,
+                             const uint8_t* compare_data, std::size_t length) {
+    // FIXME (aw): what general size to take here?
```

Comment:
Probably expand this as soon as we encounter larger streams.

Reply:
do the encode functions have a error code for "buffer too small"?

Reply:
> do the encode functions have a error code for "buffer too small"?

Yes, `EXI_ERROR__BITSTREAM_OVERFLOW`.